### PR TITLE
Feature/#16 single log line

### PR DESCRIPTION
**Pull request by ruud**
single log submit protocol plus it test

### CODE REVIEW

File: src/main/scala/performanceanalysis/logreceiver/LogReceiver.scala [hunk 1]
```diff
@@ -24,8 +30,26 @@ class LogReceiver extends Server  {
...
   }
 
   override def routes: Route = componentsRoute
+
+  private def handlePost(resultFuture: Future[Any]): Future[HttpResponse] = {
+    resultFuture.flatMap {
+      case LogSubmitted(componentId, logs) =>
+        Future(HttpResponse(status = StatusCodes.Accepted))
+      case LogParserNotFound(componentId) =>
+        ???
```
Comment (TimSoethout) [inline src/main/scala/performanceanalysis/logreceiver/LogReceiver.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/scala-academy/performance-analysis/file/2462365447/issues/source?bid=3220915&fileBranchId=3255667#l51)

File: src/main/scala/performanceanalysis/logreceiver/LogReceiver.scala [hunk 2]
```diff
@@ -24,8 +30,26 @@ class LogReceiver extends Server  {
...
   }
 
   override def routes: Route = componentsRoute
+
+  private def handlePost(resultFuture: Future[Any]): Future[HttpResponse] = {
+    resultFuture.flatMap {
+      case LogSubmitted(componentId, logLine) =>
+        Future(HttpResponse(status = StatusCodes.Accepted))
+      case LogParserNotFound(componentId) =>
+        ???
```
Comment (TimSoethout) [inline src/main/scala/performanceanalysis/logreceiver/LogReceiver.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/scala-academy/performance-analysis/file/2462588214/issues/source?bid=3220915&fileBranchId=3255667#l51)

Comment (ruud) [inline src/main/scala/performanceanalysis/logreceiver/LogReceiver.scala]: Old operator! To be implemented in an other story

File: src/main/scala/performanceanalysis/logreceiver/LogReceiver.scala [hunk 3]
```diff
@@ -24,8 +29,26 @@ class LogReceiver extends Server  {
...
```
Comment (TimSoethout) [inline src/main/scala/performanceanalysis/logreceiver/LogReceiver.scala]: Codacy Issue found: [Usage of ??? operator](https://www.codacy.com/app/scala-academy/performance-analysis/file/2462660216/issues/source?bid=3220915&fileBranchId=3255667#l50)

Comment (ruud) [inline src/main/scala/performanceanalysis/logreceiver/LogReceiver.scala]: Also an old operator! To be implemented in an other story, I guess

**Comment by ruud** [comment]
Help wanted for codacy issues, as I am going on a holiday

File: src/main/scala/performanceanalysis/logreceiver/LogReceiver.scala (hunk 3 above)
Comment (ruud) [inline src/main/scala/performanceanalysis/logreceiver/LogReceiver.scala]: Also an old operator! To be implemented in an other story, I guess

File: src/main/scala/performanceanalysis/LogParserActor.scala [hunk 4]
```diff
@@ -13,19 +15,35 @@ object LogParserActor {
...
+      .filter(_.groupCount >= 1)
+      .foldLeft(List[String]())((acc, regMatch) => regMatch.group(1) :: acc)
   }
 
+  private def parseLogLine(metrics: List[Metric], logLine: String, parsedLogs: Map[String, List[Any]]) = {
+    metrics foreach { metric =>
+      val key = s"${metric.metricKey}-$logLine"
+      val pattern: Regex = metric.regex.r
+      val list = findAllMatchedValues(pattern, logLine)
+      context.become(normal(metrics, parsedLogs.updated(key, list)))
```
Comment (geerdink) [inline src/main/scala/performanceanalysis/LogParserActor.scala:46]: Move context.become statement outside of foreach loop

Comment (TimSoethout) [inline src/main/scala/performanceanalysis/LogParserActor.scala:46]: `parsedLogs.updated(key,list)` wil be overwritten if `metrics` contains multiple items

File: src/main/scala/performanceanalysis/LogParserActor.scala [hunk 5]
```diff
@@ -13,19 +15,35 @@ object LogParserActor {
...
+  def receive: Receive = normal(Nil, Map())
 
-  def normal(metrics: List[Metric]): Receive = {
+  def normal(metrics: List[Metric], parsedLogs: Map[String, List[Any]]): Receive = {
     case RequestDetails =>
       log.debug("received request for details")
       sender ! Details(metrics)
 
+    case LogSubmitted(_, logLine) =>
+      log.debug("received log line '{}' to be parsed", logLine)
```
Comment (geerdink) [inline src/main/scala/performanceanalysis/LogParserActor.scala:26]: change to SubmitLog ??

File: src/main/scala/performanceanalysis/administrator/AdministratorActor.scala [hunk 6]
```diff
@@ -60,7 +60,7 @@ class AdministratorActor(logReceiverActor: ActorRef) extends Actor with ActorLog
     }
   }
 
-  private def handleGetDetails(logParserActors: Map[String, ActorRef], componentId: String, sender: ActorRef) = {
+  private def handleGetMetrics(logParserActors: Map[String, ActorRef], componentId: String, sender: ActorRef) = {
     routeToLogParser(logParserActors, componentId, sender) { ref =>
       log.debug(s"Requesting details from ${ref.path}")
```
Comment (geerdink) [inline src/main/scala/performanceanalysis/administrator/AdministratorActor.scala:65]: use {} instead of s...

**Pull request closed without merging**